api/search/views.py

```diff
@@ -0,0 +1,107 @@
+# -*- coding: utf-8 -*-
+
+from rest_framework import generics, permissions as drf_permissions
+from rest_framework.exceptions import ValidationError
+
+from api.base import permissions as base_permissions
+from api.base.views import JSONAPIBaseView
+from api.base.pagination import SearchPagination
+from api.files.serializers import FileSerializer
+from api.nodes.serializers import NodeSerializer
+from api.registrations.serializers import RegistrationSerializer
+from api.search.serializers import SearchSerializer
+from api.users.serializers import UserSerializer
+
+from framework.auth.core import User
+from framework.auth.oauth_scopes import CoreScopes
+
+from website.files.models import FileNode
+from website.models import Node
+from website.search import search
+from website.search.exceptions import MalformedQueryError
+from website.search.util import build_query
+
+
+class BaseSearchView(JSONAPIBaseView, generics.ListAPIView):
+
+    required_read_scopes = [CoreScopes.SEARCH]
+    required_write_scopes = [CoreScopes.NULL]
+
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+    )
+
+    pagination_class = SearchPagination
+
+    def __init__(self):
+        super(BaseSearchView, self).__init__()
+        self.doc_type = getattr(self, 'doc_type', None)
+
+    def get_queryset(self):
+        query = self.request.query_params.get('q', '*')
+        page = int(self.request.query_params.get('page', '1'))
+        start = (page - 1) * 10
```

Comment:
Page size seems like something that could change. This probably either is configured somewhere or it should be configurable. #noMagicNumbers